### PR TITLE
Review branch

### PR DESCRIPTION
**Pull request by jordaninthewind**
Updated error response to return JSON format, updated body-parser to use .json().

### CODE REVIEW

File: src/routes/routes.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+});
+
+router
+  .get(
+    '/users/:id',
+    (req, res) => {
+      const user = controller.getUser(req.params.id);
+
+      if (user) {
+        res.json({ data: user });
```
Comment (MurDaD) [inline src/routes/routes.js]: res.json(user);

File: src/routes/routes.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      res.json({ message: 'User deleted successfully.' });
+      return;
+    }
+
+    res.status(404).json({ message: 'User not found to delete.' });
+  })
+  .get('/users', (req, res) => {
+    const users = controller.showAllUsers();
+    if (users.length > 0) {
+      res.json({ data: users });
```
Comment (MurDaD) [inline src/routes/routes.js]: res.json(user);

File: src/routes/routes.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    res.status(400).json({ message: 'No users found.' });
+  })
+  .post(
+    '/users',
+    userValidator.body(querySchema),
+    (req, res) => {
+      const { login, password, age } = req.body;
+      const user = controller.createUser(login, password, age);
+      if (user) {
+        res.json({ message: 'User created.', data: user.id });
```
Comment (MurDaD) [inline src/routes/routes.js]: id: user.id

File: src/routes/routes.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+      if (user) {
+        res.json({ data: user });
+        return;
+      }
+
+      res.status(404).json({ message: 'A user with this ID was not found.' });
+    },
+  )
+  .put(
```
Comment (MurDaD) [inline src/routes/routes.js]: User should be able to update login only, or age only, now all 3 fields are required

File: src/routes/routes.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        res.json({ message: 'User created.', data: user.id });
+        return;
+      }
+
+      res.status(500).json({ message: 'User not created. Please try again.' });
+    },
+  )
+
+  .get('/search', (req, res) => {
+    const suggestedUsers = controller.getAutoSuggestUsers(req.body.substring, req.body.limit);
```
Comment (MurDaD) [inline src/routes/routes.js]: this should not be a `req.body.*`, but request params
Request example: `GET http://localhost:8080/users/search?substring=test&limit=10`

File: src/routes/routes.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+    if (suggestedUsers.length) {
+      res.json({ data: suggestedUsers });
```
Comment (MurDaD) [inline src/routes/routes.js]: res.json(suggestedUsers);

**Comment by jordaninthewind** [comment]
Updated all.

**Pull request closed without merging**